File: solutions/javascript/captains-log/1/captains-log.js
// @ts-check

/**
 * Generates a random starship registry number.
 *
 * @returns {string} the generated registry number.
 */

export function randomShipRegistryNumber() {
  const prefix = "NCC-";
  const min = 1000;
  const max = 9999;
  const number = Math.floor(Math.random() * (max - min + 1)) + min;
  return `${prefix}${number}`;
}


/**
 * Generates a random stardate.
 *
 * @returns {number} a stardate between 41000 (inclusive) and 42000 (exclusive).
 */
export function randomStardate() {
  const min = 41000.0;
  const max = 42000.0;
  return min + Math.random() * (max - min);
}


/**
 * Generates a random planet class.
 *
 * @returns {string} a one-letter planet class.
 */
export function randomPlanetClass() {
  const classes = ['D', 'H', 'J', 'K', 'L', 'M', 'N', 'R', 'T', 'Y'];
  const index = Math.floor(Math.random() * classes.length);
  return classes[index];
}

File: solutions/java/captains-log/1/src/main/java/CaptainsLog.java
import java.util.Random;

class CaptainsLog {

    private static final char[] PLANET_CLASSES = new char[]{'D', 'H', 'J', 'K', 'L', 'M', 'N', 'R', 'T', 'Y'};

    private Random random;

    CaptainsLog(Random random) {
        this.random = random;
    }

    char randomPlanetClass() {
        int idx = random.nextInt(PLANET_CLASSES.length);
        return PLANET_CLASSES[idx];
    }

    String randomShipRegistryNumber() {
        int number = 1000 + random.nextInt(9000); // 1000..9999 inclusive
        return "NCC-" + number;
    }

    double randomStardate() {
        return 41000.0 + 1000.0 * random.nextDouble();
    }
}